third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -976,11 +1027,27 @@ impl IndexedRef {
 
 impl Value {
     pub fn equals(&self, other: &Self) -> PartialVMResult<bool> {
-        self.0.equals(&other.0)
+        // Note: no need to feature gate because values were previously bounded at pack time.
+        self.0
+            .equals(&other.0, 1, Some(DEFAULT_MAX_VM_VALUE_NESTED_DEPTH))
```

Comment:
If this is based on a constant, should there perhaps an assertion somewhere that the VM config value is <=? As a constant here, this value should be interpreted as the maximum we could ever allow, while the vm config may further constraint for other contexts.